articles/115_topic_and_service_name_mapping.md

```diff
@@ -0,0 +1,424 @@
+Since ROS 2 topic and service names are expanded to fully qualified names, any balanced bracket (`{}`) substitutions and tildes (`~`) will have been expanded.
+Additionally any URL related syntax, e.g. the `rostopic://` prefix, will be removed once parsed.
+Therefore only forward slashes have to be substituted when converting to DDS topic names.
+
+### Substitution of the Namespace Delimiter
+
+The namespace delimiter in ROS 2 topic and service names, a forward slash (`/`), will be replaced with double underscores (`__`).
+Note that as fully qualified ROS 2 topic and service names are absolute, there is always a leading forward slash (`/`).
+
+### ROS Specific Name Prefix
+
+In order to differentiate ROS topics easily, all DDS topic names created by ROS shall be prefixed with `rX`, where `X` is a single character that indicates to which subsystem of ROS the topic belongs.
+For example, a plain topic called `/foo` would translate to a DDS topic named `rt__foo`, which is the result of concatenating the prefix `rt` for being a ROS topic, with `__` for the leading `/`, and the topic name `foo`.
+
+For systems where Services are implemented with topics (like with OpenSplice), a different subsystem character can be used: `rq` for the request topic and `rr` for the response topic.
+On systems where the implementation is handled for us by DDS (like with Connext), we use `rs` as the common prefix.
+
+Here is a non-exhaustive list of prefixes:
+
+| ROS Subsystem        | Prefix |
+|----------------------|--------|
+| ROS Topics           | rt     |
+| ROS Service Request  | rq     |
+| ROS Service Response | rr     |
+| ROS Service          | rs     |
+| ROS Parameter        | rp     |
+| ROS Action           | ra     |
+
+While all planned prefixes consist of two characters, i.e. `rX`, anything proceeding the first namespace separator, i.e. `__`, can be considered part of the prefix.
+The standard reserves the right to use up to 8 characters for the prefix in case additional prefix space is needed in the future.
+
+### ROS to DDS Name Conversion Examples
+
+Here are some examples of translations between ROS topic names and the corresponding DDS topic names:
+
+| ROS Name          | Subsystem | DDS Name               |
+|-------------------|-----------|------------------------|
+| `/foo`            | Topic     | `rt__foo`              |
+| `/foo/bar`        | Topic     | `rt__foo__bar`         |
+| `/_foo/bar`       | Topic     | `rt___foo__bar`        |
+| `/foo/_bar`       | Topic     | `rt__foo___bar`        |
+| `/_foo/_bar`      | Topic     | `rt___foo___bar`       |
+| `/_foo/_bar/_baz` | Topic     | `rt___foo___bar___baz` |
+
+### ROS Topic and Service Name Length Limit
+
+Because DDS topic names must be limited to 255 characters, the length of a ROS topic or service name is also limited in length.
+Calculating this length is governed by an unfortunately non-trivial algorithm.
```

Comment:
Ah, that's probably simpler, do you want to reword it or would you prefer I do it?